tutor/commands/plugins.py

```diff
@@ -153,22 +153,21 @@ def enable(context: Context, plugin_names: list[str]) -> None:
 @click.argument(
     "plugin_names", metavar="plugin", nargs=-1, type=PluginName(allow_all=True)
 )
-@click.pass_obj
-def disable(context: Context, plugin_names: list[str]) -> None:
-    config = tutor_config.load_minimal(context.root)
+@click.pass_context
+def disable(context: click.Context, plugin_names: list[str]) -> None:
+    config = tutor_config.load_minimal(context.obj.root)
     disable_all = "all" in plugin_names
     disabled: list[str] = []
     for plugin in tutor_config.get_enabled_plugins(config):
         if disable_all or plugin in plugin_names:
             fmt.echo_info(f"Disabling plugin {plugin}...")
-            hooks.Actions.PLUGIN_UNLOADED.do(plugin, context.root, config)
+            hooks.Actions.PLUGIN_UNLOADED.do(plugin, context.obj.root, config)
             disabled.append(plugin)
             fmt.echo_info(f"Plugin {plugin} disabled")
     if disabled:
-        tutor_config.save_config_file(context.root, config)
-        fmt.echo_info(
-            "You should now re-generate your environment with `tutor config save`."
-        )
+        ENV_PATCHES_DICT.clear()
```

Comment:
Not quite :) One of the advantages of having an event-driven API is that we don't need to import dependencies. In particular, we don't want to spread the use of `ENV_PATCHES_DICT` outside of its own module (tutor.plugins). So, we should try not to use that filter here.

Instead, we should add a callback to the PLUGIN_UNLOADED action. That callback will clear and re-compute the ENV_PATCHES_DICT. It will be called automatically every time a plugin is unloaded. Thus, we don't have to worry about it outside of its own module.

This is a bit difficult to explain... Am I making sense?

Reply:
Yes. Totally. You mean something like this:

``` python
@hooks.Actions.PLUGIN_UNLOADED.add()
def _clear_plugin_patches(plugin: str, root: str, _config: Config) -> None:
    """
    After a plugin is disabled, we clear the patches.
    """
    ENV_PATCHES_DICT.clear()
```

Reply:
Yes. And the callback should also call a cache-filling function.

Reply:
I apologize, but I'm feeling lost again. The issue was resolved after I cleared the `ENV_PATCHES_DICT`, and the disable command is now functioning properly. If by "cache-filling" you mean executing `.iterate()` and updating the `ENV_PATCHES_DICT`, I believe that `_update_enabled_plugins_on_unload` will take care of that. However, if I am mistaken, I would appreciate some assistance.

Reply:
Looks like I need to spend more time on tutor codebase :)